tests/testthat/testEdgeThresh.R

```diff
@@ -0,0 +1,251 @@
+# edgeThresh.R
+#
+# Structure your tests according to the three principles expressed below:
+#    test_that("parameter errors are correctly handled", { ... })
+#    test_that("a sane input gives an expected output", { ... })
+#    test_that("a corrupt input does not lead to corrupted output", { ...})
+
+
+context("Graph permutation and delta calculation")
+
+
+# ==== BEGIN SETUP AND PREPARE =================================================
+# OLOG <- as.character(getOption("rete.logfile"))   # save original logfile name
+# logFileName(fPath = tempdir(), setOption = TRUE)  # make tempdir() the log dir
+# NL <- .PlatformLineBreak()
+# if (file.exists(logName)) { file.remove(logName)}
+
+# ((Is rete.EGGprototype only the barebones valid EGG, or does it contain testable
+# data?))
+myEGG <- rete.EGGprototype
+nLargest <- max(igraph::components(EGG)$csize)
+permutedGraph <- .permuteGraph(EGG)
+nVertices <- igraph::vcount(EGG)
+nEdges <- igraph::ecount(EGG)
+# ==== END SETUP AND PREPARE ===================================================
+
+
+test_that("edgeThresh parameter errors are correctly handled", {
+    # Try each parameter missing in turn (only EGG has no defaults)
+    expect_error(edgeThresh())
+
+    # Try parameters out of expected bounds
+    expect_error(edgeThresh(myEGG, Lmax =  1))
+    expect_error(edgeThresh(myEGG, Lmax =  0))
+    expect_error(edgeThresh(myEGG, Lmax = -1))
+    expect_error(edgeThresh(myEGG, Lmax = nLargst - 1))
+
+    expect_error(edgeThresh(myEGG, Lmax = 20, N =  0))
+    expect_error(edgeThresh(myEGG, Lmax = 20, N = -1))
+
+    expect_error(edgeThresh(myEGG, Lmax = 20, Q =  0))
+    expect_error(edgeThresh(myEGG, Lmax = 20, Q = -1))
+
+
+    # Try NULL and length-zero parameters: be sure none lead to an erroneous
```

Comment:
While that's certainly true of input parameters, this consideration is more important for the code itself - however this can't be easily checked here because you are already assuming the parameters can't go out of range.